src/Storages/Hive/StorageHive.cpp

```diff
@@ -115,19 +116,31 @@ class StorageHiveSource : public SourceWithProgress, WithContext
         , format(std::move(format_))
         , compression_method(compression_method_)
         , max_block_size(max_block_size_)
-        , sample_block(std::move(sample_block_))
+        , header_block(header_block_)
+        , sample_block(sample_block_)
         , to_read_block(sample_block)
         , columns_description(getColumnsDescription(sample_block, source_info))
         , text_input_field_names(text_input_field_names_)
         , format_settings(getFormatSettings(getContext()))
     {
         /// Initialize to_read_block, which is used to read data from HDFS.
-        to_read_block = sample_block;
         for (const auto & name_type : source_info->partition_name_types)
         {
             to_read_block.erase(name_type.name);
         }
-
+        // hive table header doesn't contain partition columns, we need at least one column to read.
+        if (!to_read_block.columns())
+        {
+            all_to_read_are_partition_columns = true;
```

Comment:
why not just assign to sample_block in this case

Reply:
that means all columsn in the sample_block are the partition columns, arrow table's columns doesn't include the partition columns. It would fail to read with only partition columns.